### DISCUSSION TITLE
typedef neorv32_uart0_t/neorv32_uart1_t change to neorv32_uart_t

### DISCUSSION THREAD

**Discussion by andkae** [stnolting/neorv32#492] [General]
Hi Stephan

at the moment i start to work on an IRQ driven UART driver. In this context i saw that in

https://github.com/stnolting/neorv32/blob/d01da1ffbb22765876814e40f1e26cdd720df0a3/sw/lib/include/neorv32.h#L1029-1044

for UART0 and UART1 the same register layout, but two different typedefs are used. Is there a special reason to do this? Or could also look the implementation like follows:

```c
typedef struct __attribute__((packed,aligned(4))) {
  uint32_t CTRL;  /**< offset 0: control register (#NEORV32_UART_CTRL_enum) */
  uint32_t DATA;  /**< offset 4: data register (#NEORV32_UART_DATA_enum) */
} neorv32_uart_t;

/** UART0 module base address */
#define NEORV32_UART0_BASE (0xFFFFFFA0U)
/** UART0 module hardware access (#neorv32_uart_t) */
#define NEORV32_UART0 (*((volatile neorv32_uart_t*) (NEORV32_UART0_BASE)))

/** UART1 module base address */
#define NEORV32_UART1_BASE (0xFFFFFFD0U)
/** UART1 module hardware access (#neorv32_uart_t) */
#define NEORV32_UART1 (*((volatile neorv32_uart_t*) (NEORV32_UART1_BASE)))
```

What do you think?

BR,
Andreas

**Comment by stnolting** [comment]
Hey there!

You are right, this is entirely redundant. Would you open a PR to clean this up? :wink:

**Reply by akaeba** [reply]
Done in #493.
